mlflow/utils/databricks_utils.py

```diff
@@ -506,37 +542,67 @@ def get_databricks_host_creds(server_uri=None):
             requests.
 
     Returns:
-        MlflowHostCreds which includes the hostname and authentication information necessary to
+        MlflowHostCreds which includes the hostname if databricks sdk authentication is available,
+        otherwise includes the hostname and authentication information necessary to
         talk to the Databricks server.
     """
-    profile, path = get_db_info_from_uri(server_uri)
-    config = ProfileConfigProvider(profile).get_config() if profile else get_config()
-    # if a path is specified, that implies a Databricks tracking URI of the form:
-    # databricks://profile-name/path-specifier
-    if (not config or not config.host) and path:
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
 
-    insecure = hasattr(config, "insecure") and config.insecure
+    from databricks.sdk import WorkspaceClient
```

Comment:
Our test requirements specify version 0.20.0, which is only installed in 15.x runtime, and this SDK was introduced into DBR in 13.x (with version 0.1.6). 
Older runtimes will exhibit a breaking change here with authentication in runtime if a user installs MLflow 2.14.0. 
Can we wrap this import in a try catch and if the import fails, utilize the fallback logic that is preserved here? 

Please test an install of this branch on both MLR 10LTS and 13LTS to ensure that auth functions correctly for: 
1. databricks-sdk is not in the runtime environment
2. databricks-sdk < version 0.20.0 

To verify that fallback logic for these older runtimes functions without any breaking changes.

Reply:
(The main condition that I'd like to verify is if databricks-sdk being upgraded to 0.20.0 in these older runtimes causes any issues, since installing MLflow will force upgrade or simply install the package). I'm not sure if the runtime environment exposes everything needed to make the SDK function correctly.

Reply:
Confirmed:

(1) when installing MLflow or MLflow skinny , the databricks-sdk >= 0.20 is forcily installed dependency. So it wouldn't cause importing error.

(2) I tested in databricks runtime 9.3 / 10.3 / 13.3 ,  databricks-sdk >= 0.20 is compatible with them. sdk team will ensure compatibility for runtime versions in lifecycle.